bin/DESeq2.R

```diff
@@ -173,19 +190,79 @@ if (!is.null(opt$relevel)) {
+        coldata$combfactor <- metadata$combfactor
+        rownames(coldata) <- NULL
+
+        #Do tximeta, this is necessary to run DESeq on rsem
+        se <- tximeta(coldata, type="rsem", txIn=FALSE, txOut=FALSE, skipMeta=TRUE)
+        assays(se)$length[ assays(se)$length == 0] <- NA # set these as missing
+        #Impute lengths for the 0-length values:
+        length_imp <- impute.knn(assays(se)$length)
+        assays(se)$length <- length_imp$data
+        #dds from SummarizedExperiment <se>, then run DESeq
+        cds <- DESeqDataSet(se, design = as.formula(eval(parse(text=as.character(design[[1]])))))
+        cds <- DESeq(cds)
+    } else if (opt$input_type == "salmon") {
+        files <- file.path(gsub("/$", "", path_count_table), qbicCodes, "quant.sf")
+        if (!(all(file.exists(files)))) {
+            stop("DESeq2.R could not find all of the specified quant.sf files!")
+        }
+        ## Import all of the samples information and transform the identifiers
+        txi.salmon <- tximport(files, type = "salmon", tx2gene = tx2gene_gtf, ignoreTxVersion = T)
+        # Run cds with tximport depending on whether rsem or salmon was used
+                #Extract condition columns and other info for tximeta
+        condition_names <- unlist(strsplit(design[,1], split = " "))
+        condition_names <- grep("condition", condition_names, value=T)
+        sampleconditions <- data.frame(metadata[,condition_names])
+        colnames(sampleconditions) <- condition_names
+        coldata <- data.frame(files = files, names= qbicCodes, sampleconditions)
+        coldata$combfactor <- metadata$combfactor
+        rownames(coldata) <- qbicCodes
+        cds <- DESeqDataSetFromTximport(txi=txi.salmon, colData =coldata, design = eval(parse(text=as.character(design[[1]]))))
+        cds <- DESeq(cds)
+    }
+} else {
+    stop("Input type must be one of [rawcounts, rsem, salmon]!")
+}
 
 # SizeFactors(cds) as indicator of library sequencing depth
 write.table(sizeFactors(cds),paste("differential_gene_expression/gene_counts_tables/sizeFactor_libraries.tsv",sep=""), append = FALSE, quote = FALSE, sep = "\t",eol = "\n", na = "NA", dec = ".", row.names = T,  col.names = F, qmethod = c("escape", "double"))
 
 # Write cds assay table to file
 write.table(counts(cds, normalized=T), paste("differential_gene_expression/gene_counts_tables/deseq2_table.tsv", sep=""), append=F, quote = F, sep = "\t", eol = "\n", na = "NA", dec = ".", row.names = T, col.names = T, qmethod = c("escape", "double"))
-
-# Write raw counts to file
-count_table_names <- merge(x=gene_names, y=count.table, by.x = "Ensembl_ID", by.y="row.names")
-write.table(count_table_names, paste("differential_gene_expression/gene_counts_tables/raw_gene_counts.tsv",sep=""), append = FALSE, quote = FALSE, sep = "\t",eol = "\n", na = "NA", dec = ".", row.names = F, qmethod = c("escape", "double"))
-
+if (opt$input_type == "rawcounts"){
```

Comment:
Here it is important to provide the raw counts for some customers. So the salmon and rsem raw counts should also be copied to the output folder `gene_counts_table`

Reply:
Done!